src/USER-MESONT/pair_mesont_tpm.cpp

```diff
@@ -508,58 +520,58 @@ void PairMESONTTPM::compute(int eflag, int vflag){
-  if (atom->nmax > maxeatom) {
-    maxeatom = atom->nmax;
-    memory->destroy(eatom);
-    memory->create(eatom,comm->nthreads*maxeatom,"pair:eatom");
-    memory->destroy(eatom_s);
-    memory->create(eatom_s,comm->nthreads*maxeatom,"pair:eatom_s");
-    memory->destroy(eatom_b);
-    memory->create(eatom_b,comm->nthreads*maxeatom,"pair:eatom_b");
-    memory->destroy(eatom_t);
-    memory->create(eatom_t,comm->nthreads*maxeatom,"pair:eatom_t");
+  //convert from sorted representation
+  for (int i = 0; i < nall; i++){
+      int idx = ntlist.get_idx(i);
+      for (int j = 0; j < 3; j++) f[idx][j] += f_sort[3*i+j];
+      buckling[idx] = b_sort[i];
   }
-
-  if (atom->nmax > maxvatom) {
-    maxvatom = atom->nmax;
-    memory->destroy(vatom);
-    memory->create(vatom,comm->nthreads*maxvatom,6,"pair:vatom");
+  if(eflag){
+    eng_vdwl = 0.0; energy_s = 0.0;
+    energy_b = 0.0; energy_t = 0.0;
+    for (int i = 0; i < ntot; i++){
+      eatom[i] = 0.0; eatom_s[i] = 0.0;
+      eatom_b[i] = 0.0; eatom_t[i] = 0.0;
+    }
+    for (int i = 0; i < nall; i++){
+      int idx = ntlist.get_idx(i);
+      eatom_s[idx] = u_ts_sort[i];
+      eatom_b[idx] = u_tb_sort[i];
+      eatom_t[idx] = u_tt_sort[i];
+      eatom[idx] = u_ts_sort[i] + u_tb_sort[i] + u_tt_sort[i];
+      energy_s += u_ts_sort[i];
+      energy_b += u_tb_sort[i];
+      energy_t += u_tt_sort[i];
+    }
+    eng_vdwl = energy_s + energy_b + energy_t;
   }
-
-  // zero accumulators
-  eng_vdwl = 0.0; energy_s = 0.0;
-  energy_b = 0.0; energy_t = 0.0;
-  for (int i = 0; i < 6; i++) virial[i] = 0.0;
-  for (int i = 0; i < ntot; i++){
-    eatom[i] = 0.0; eatom_s[i] = 0.0;
-    eatom_b[i] = 0.0; eatom_t[i] = 0.0;
+  if(vflag){
+    for (int i = 0; i < 6; i++) virial[i] = 0.0;
+    for (int i = 0; i < nall; i++){
+      int idx = ntlist.get_idx(i);
+      virial[0] += s_sort[9*i+0]; //xx
+      virial[1] += s_sort[9*i+4]; //yy
+      virial[2] += s_sort[9*i+8]; //zz
+      virial[3] += s_sort[9*i+1]; //xy
+      virial[4] += s_sort[9*i+2]; //xz
+      virial[5] += s_sort[9*i+5]; //yz
+    }
   }
-  for (int i = 0; i < ntot; i++)
-    for (int j = 0; j < 6; j++) vatom[i][j] = 0.0;
-
-  //convert from sorted representation
-  for (int i = 0; i < nall; i++){
-    int idx = ntlist.get_idx(i);
-    for (int j = 0; j < 3; j++) f[idx][j] += f_sort[3*i+j];
-    eatom_s[idx] = u_ts_sort[i];
-    eatom_b[idx] = u_tb_sort[i];
-    eatom_t[idx] = u_tt_sort[i];
-    eatom[idx] = u_ts_sort[i] + u_tb_sort[i] + u_tt_sort[i];
-    energy_s += u_ts_sort[i];
-    energy_b += u_tb_sort[i];
-    energy_t += u_tt_sort[i];
-    vatom[idx][0] = s_sort[9*i+0]; //xx
-    vatom[idx][1] = s_sort[9*i+4]; //yy
-    vatom[idx][2] = s_sort[9*i+8]; //zz
-    vatom[idx][3] = s_sort[9*i+1]; //xy
-    vatom[idx][4] = s_sort[9*i+2]; //xz
-    vatom[idx][5] = s_sort[9*i+5]; //yz
-    for (int j = 0; j < 6; j++) virial[j] += vatom[idx][j];
-    buckling[idx] = b_sort[i];
+  int vflag_atom = vflag & 4;
```

Comment:
this should not be needed and is shadowing an existing variable. vflag_atom (and vflag_global) are set by calling ev_init().

Reply:
Thank you, I'll fix it.